test/resource_test.rb

```diff
@@ -133,6 +176,119 @@ def test_acquire_with_fork
     end
   end
 
+  def test_quota_acquire
+    quota = 0.5
+    workers = 9
+
+    resource = create_resource :testing, quota: quota, timeout: 0.1
+    fork_workers(count: workers, quota: quota, wait_for_timeout: true)
+
+    # Ensure that the number of tickets is correct and none are remaining
+    assert_equal quota * (workers + 1), resource.tickets
+    assert_equal 0, resource.count
+
+    # Ensure that no more tickets may be allocated
+    assert_raises Semian::TimeoutError do
+      resource.acquire {}
+    end
+
+    signal_workers('TERM')
+
+    # Ensure the correct number of processes timed out
+    timeouts = Process.waitall.count { |s| s.last.exitstatus == 100 }
+    assert_equal workers - ((1 - quota) * workers).ceil, timeouts
+
+    # Ensure that the tickets were released
+    assert_equal resource.tickets, resource.count
+  end
+
+  def test_quota_increase
+    quota = 0.5
+    new_quota = 0.7
+    workers = 50
+    added_workers = 50
+
+    fork_workers(count: workers, quota: quota, wait_for_timeout: true)
+    fork_workers(count: added_workers, quota: new_quota, wait_for_timeout: true)
+
+    # Allow all processes to terminate
+    signal_workers('TERM')
+    es = Process.waitall
+
+    # Ensure the correct number of processes timed out, based on the *new* quota
+    actual_timeouts = es.count { |s| s.last.exitstatus == 100 }
+    expected_timeouts = ((1 - new_quota) * (workers + added_workers)).floor
+    assert_equal expected_timeouts, actual_timeouts
+  end
+
+  def test_quota_decrease
+    quota = 0.5
+    new_quota = 0.3
+    workers = 50
+
+    # Spawn some workers with an initial quota
+    fork_workers(count: workers - 1, quota: quota, wait_for_timeout: true) do
+      sleep 2 # allow them to stay registered briefly after being signalled
+    end
+
+    signal_workers('TERM')
+    resource = create_resource :testing, quota: new_quota, timeout: 0.1
+
+    assert_equal((workers * new_quota).ceil, resource.tickets)
+  end
+
+  def test_quota_sets_tickets_from_workers
+    quota = 0.5
+    workers = 50
+
+    resource = create_resource :testing, quota: quota, timeout: 0.1
+    fork_workers(count: workers - 1, quota: quota, wait_for_timeout: true)
```

Comment:
This is because of the worker that gets registered for the test process itself (the line above). To make the math simpler, we subtract that from the total number of workers to fork.

Is there another way you can think of to do this more intuitively?